### PR TITLE
remove duplicate copy of prefetching code

### PR DESCRIPTION
**Pull request by lbergelson**
Removing our copy of the prefetching code and using the copy bundled with the nio library instead.  Adding two new methods to BucketUtils to make it less verbose to create them and to re-wrap an IOException into the expected GATKException.

### CODE REVIEW

File: src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java [hunk 1]
```diff
@@ -429,4 +433,26 @@ public static CloudStorageConfiguration getCloudStorageConfiguration(int maxReop
...
+        }
+    }
+
+    /**
+     * Creates a wrapping function which adds a prefetcher if the buffer size is > 0 if its <= 0 then this wrapper returns the
+     * original channel.
+     * @param cloudPrefetchBuffer the prefetcher buffer size in MB
+     */
+    public static Function<SeekableByteChannel, SeekableByteChannel> getPrefetchingWrapper(final int cloudPrefetchBuffer) {
+        return cloudPrefetchBuffer > 0 ? is -> addPrefetcher(cloudPrefetchBuffer, is) : Utils.identityFunction();
```
[the remark this answers is not in the document]
Comment (lbergelson) [reply src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java]: that's a good question.... I don't remember where is came from...

[the remark this answers is not in the document]
Comment (davidbenjamin) [reply src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java]: It depends on what the meaning of "is" is.

[the remark this answers is not in the document]
Comment (lbergelson) [reply src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java]: `rawChannel` is a good option.

File: src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java [hunk 2]
```diff
@@ -429,4 +433,26 @@ public static CloudStorageConfiguration getCloudStorageConfiguration(int maxReop
...
+    public static SeekableByteChannel addPrefetcher(int bufferSizeMB, SeekableByteChannel channel) {
+        try {
+            return SeekableByteChannelPrefetcher.addPrefetcher(bufferSizeMB, channel);
+        } catch (final IOException ex) {
+            throw new GATKException("Unable to initialize the prefetcher: " + ex);
+        }
+    }
+
+    /**
+     * Creates a wrapping function which adds a prefetcher if the buffer size is > 0 if its <= 0 then this wrapper returns the
```
[the remark this answers is not in the document]
Comment (lbergelson) [reply src/main/java/org/broadinstitute/hellbender/utils/gcs/BucketUtils.java]: done

File: src/main/java/org/broadinstitute/hellbender/utils/nio/ReadsIterable.java [hunk 3]
```diff
@@ -35,7 +35,7 @@
     private final boolean removeHeader = true;
 
     class ReadsIterator implements CloseableIterator<SAMRecord> {
-        private final static int BUFSIZE = 200 * 1024 * 1024;
+        private final static int BUFSIZE_MB = 200;
```
[the remark this answers is not in the document]
Comment (lbergelson) [reply src/main/java/org/broadinstitute/hellbender/utils/nio/ReadsIterable.java]: done, although we should delete that class in a separate pr

File: src/main/java/org/broadinstitute/hellbender/utils/nio/ReadsIterable.java [hunk 4]
```diff
@@ -47,8 +47,7 @@ public ReadsIterator() throws IOException {
             SeekableStream indexInMemory = new ByteArraySeekableStream(indexData);
             // set high-level retries to deal with servers that might be temporarily overloaded
             // while we're reading a very long file from them.
-            SeekableByteChannelPrefetcher chan = new SeekableByteChannelPrefetcher(
-                Files.newByteChannel(fpath), BUFSIZE);
+            SeekableByteChannel chan = BucketUtils.addPrefetcher( BUFSIZE_MB, Files.newByteChannel(fpath));
```
[the remark this answers is not in the document]
Comment (lbergelson) [reply src/main/java/org/broadinstitute/hellbender/utils/nio/ReadsIterable.java]: done

**Pull request merged**